### DISCUSSION TITLE
Abaqus mesh reader questions

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [libMesh/libmesh#3259] [Q&A] [answered]
Hi all -

I'd like to read in an Abaqus mesh (.inp file) but am having issues. It contains 8-noded quads and 6-noded triangles. From poking around in the [AbaqusIO class](https://github.com/libMesh/libmesh/blob/devel/src/mesh/abaqus_io.C), I think I see the issue, but allow me to ask questions to make sure I understand:

1. I noticed there is currently no support for what I would call "higher order" 2D elements - TRI6 and QUAD8 for example. Is there a reason for this or could this be added?

2. I see there are node and side maps in the AbaqusIO class for converting the element ordering from Abaqus to libmesh. I see how the nodes are numbered, for instance, [here](https://github.com/libMesh/libmesh/blob/devel/include/geom/face_quad8.h) for QUAD8, but how are the sides numbered?

**Comment by jwpeterson** [comment, marked as answer]
1. There's no fundamental reason for having ~no~ missing support for second-order elements in `AbaqusIO`. The reader class was developed by "reverse engineering" based on example input files... there may be an Abaqus input file specification out there somewhere, not sure.  
2. The sides are numbered according to the `side_nodes_map`, which is defined in the C-file for each element type, e.g. [Quad8](https://github.com/libMesh/libmesh/blob/devel/src/geom/face_quad8.C)

**Reply by jessecarterMOOSE** [reply]
Thanks. Given that info I was able to add some lines to libmesh to read in my Abaqus mesh. Is there any interest in a PR for this? It's maybe 10 lines. I can't share my mesh though.

**Reply by jwpeterson** [reply]
Yes, a PR would definitely be of interest. We actually don't have any testing of `AbaqusIO` at the libmesh level (might be some in MOOSE?), so I can't reasonably demand a test from you, but if you have the time, a single-element input file for testing the new element types you added support for would be great.

**Reply by jessecarterMOOSE** [reply]
Yeah there are a few tests in MOOSE which all use 3D elements. MOOSE is my intended use for this feature, but I figured I'd ask here since the mesh reader is in libmesh.

**Comment by jwpeterson** [comment]
There is a list of resources that we used to define the current set of elements in the commit message of 0a65e380ccde33257d5e6312bd9b85fa5a7006ea. Not sure if any of those are either a) still around or b) useful to you.
